src_c/base.c

```diff
@@ -533,6 +533,62 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg) {
+            PyErr_Clear();
+            PyErr_SetString(PyExc_TypeError, msg);
+        }
+        return 0;
+    }
+    return 1;
+}
+
+static int
+pg_TwoIntsFromObjEx(PyObject *obj, int *val1, int *val2, const char *msg)
+{
+    PyObject *fseqence = NULL;
+    Py_ssize_t length = 0;
+
+    if (!(fseqence = PySequence_Fast(obj, "A sequence was expected")))
+        return 0;
+
+    length = PySequence_Fast_GET_SIZE(fseqence);
+
+    if (length == 2) {
+        if (!pg_IntFromObjEx(PySequence_Fast_GET_ITEM(fseqence, 0), val1,
+                             msg) ||
+            !pg_IntFromObjEx(PySequence_Fast_GET_ITEM(fseqence, 1), val2,
+                             msg)) {
+            Py_DECREF(fseqence);
+            return 0;
+        }
+    }
+    else if (length == 1) {
+        /* Handle case of ((x, y), ) 'sequence inside sequence' */
+        PyObject *tmp = PySequence_Fast_GET_ITEM(fseqence, 0);
+        Py_DECREF(fseqence);
+        return pg_TwoIntsFromObjEx(tmp, val1, val2, msg);
+    }
+    else {
+        Py_DECREF(fseqence);
+        PyErr_SetString(PyExc_ValueError,
+                        "Invalid sequence size, must be either 1 or 2");
```

Comment:
```suggestion
        PyErr_SetString(PyExc_ValueError, msg);
```
This should be a `ValueError` aight, but I think we should still use the more descriptive caller-passed message string here